README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
+
+#### Ubuntu 16.04
+
+##### 1. Install git, vagrant, lxc, and related dependencies:
+
+```
+christie@xenial-desktop:~
+$ sudo apt-get install git vagrant lxc lxc-templates cgroup-lite redir
+```
+
+##### 2. Install the vagrant lxc plugin:
+
+```
+christie@xenial-desktop:~
+$ vagrant plugin install vagrant-lxc
+Installing the 'vagrant-lxc' plugin. This can take a few minutes...
+Installed the plugin 'vagrant-lxc (1.2.1)'!
+```
+
+If you encounter an error when trying to install the vagrant-lxc plugin, [see
+this](#nomethoderror-when-installing-vagrant-lxc-plugin-ubuntu-1604).
+
+##### 3. Configure sudo to be passwordless
+
+Finally, [configure sudo to be passwordless when using Vagrant LXC][avoiding-sudo]:
+
+```
+christie@xenial-desktop:~
+$ vagrant lxc sudoers
+[sudo] password for christie:
+```
+
+Now you are ready for [Step 2: Get Zulip Code.](#step-2-get-zulip-code)
+
+### Step 2: Get Zulip Code
+
+If you haven't already created an ssh key and added it to your Github account,
+you should do that now by following [these
+instructions](https://help.github.com/articles/generating-an-ssh-key/).
+
+1. In your browser, visit https://github.com/zulip/zulip and click the
+   `fork` button. You will need to be logged in to Github to do this.
+2. Open Terminal (OS X/Ubuntu) or Cygwin (Windows; must run as an Administrator)
+3. In Terminal/Cygwin, clone your fork:
+```
+git clone git@github.com:YOURUSERNAME/zulip.git
+```
+
+This will create a 'zulip' directory and download the Zulip code into it.
+
+Don't forget to replace YOURUSERNAME with your git username. You will see
+something like:
+
+```
+christie@win10 ~
+$ git clone git@github.com:YOURUSERNAME/zulip.git
+Cloning into 'zulip'...
+remote: Counting objects: 73571, done.
+remote: Compressing objects: 100% (2/2), done.
+remote: Total 73571 (delta 1), reused 0 (delta 0), pack-reused 73569
+Receiving objects: 100% (73571/73571), 105.30 MiB | 6.46 MiB/s, done.
+Resolving deltas: 100% (51448/51448), done.
+Checking connectivity... done.
+Checking out files: 100% (1912/1912), done.`
+```
+
+Now you are ready for [Step 3: Start the dev
+environment.](#step-3-start-the-dev-environment)
+
+### Step 3: Start the dev environment 
+
+Change into the zulip directory and tell vagrant to start the Zulip
+dev environment with `vagrant up`.
+
+```
+christie@win10 ~
+$ cd zulip
+
+christie@win10 ~/zulip
+$ vagrant up
+```
+
+The first time you run this command it will take some time because vagrant
+does the following:
+
+- downloads the base Ubuntu 14.04 virtual machine image (for OS X and Windows)
+  or container (for Ubuntu)
+- configures this virtual machine/container for use with Zulip,
+- creates a shared directory `/srv/zulip` within the virtual machine/container
+  that maps to where you have cloned the zulip code, and
+- runs the `provision.py` script, which downloads all required dependencies
+  and sets up the python environment for Zulip within the virtual
+  machine/container. 
```

Comment:
Maybe better would be "runs the `provision.py` script inside the virtual machine/container, which ... ?